contracts/vaults/RibbonThetaVault.sol

```diff
@@ -536,26 +533,60 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         address newOption = nextOption;
         require(newOption > PLACEHOLDER_ADDR, "!nextOption");
 
+        uint256 pendingAmount = totalPending();
+        uint256 currentSupply = totalSupply();
+        uint256 currentBalance = assetBalance();
+        uint256 roundStartBalance = currentBalance.sub(pendingAmount);
+
+        uint256 singleShare = 10**uint256(_decimals);
+
+        uint256 currentPricePerShare =
+            currentSupply > 0
+                ? singleShare.mul(roundStartBalance).div(currentSupply)
+                : singleShare;
+
+        // After closing the short, if the options expire in-the-money
+        // vault pricePerShare would go down because vault's asset balance decreased.
+        // This ensures that the newly-minted shares do not take on the loss.
+        uint256 mintShares =
+            currentSupply > 0
+                ? pendingAmount.mul(singleShare).div(currentPricePerShare)
+                : pendingAmount;
+
+        // Vault holds temporary custody of the newly minted vault shares
+        _mint(address(this), mintShares);
+
+        uint256 newSupply = currentSupply.add(mintShares);
+
+        // TODO: We need to use the pps of the round they scheduled the withdrawal
+        // not the pps of the new round. https://github.com/ribbon-finance/ribbon-v2/pull/10#discussion_r652174863
+        uint256 queuedWithdrawAmount =
+            newSupply > 0
+                ? queuedWithdrawShares.mul(currentBalance).div(newSupply)
+                : 0;
+
+        uint256 balanceSansQueued = currentBalance.sub(queuedWithdrawAmount);
+
+        _totalPending = PLACEHOLDER_UINT;
         currentOption = newOption;
         nextOption = PLACEHOLDER_ADDR;
-        round += 1;
+        lockedAmount = balanceSansQueued;
 
-        uint256 currentBalance = assetBalance();
-        (uint256 queuedWithdrawAmount, , ) =
-            _withdrawAmountWithShares(queuedWithdrawShares, currentBalance);
-        uint256 shortAmount = currentBalance.sub(queuedWithdrawAmount);
-        lockedAmount = shortAmount;
+        // Finalize the pricePerShare at the end of the round
+        uint16 currentRound = round;
+        roundPricePerShare[currentRound] = currentPricePerShare;
+        round = currentRound + 1;
```

Comment:
The `uint16 currentRound = round` is needed because it helps us save ~100 gas from avoiding a read

Reply:
understood